library/Rules/PostalCode.php

```diff
@@ -53,7 +53,7 @@ class PostalCode extends Regex
         "FM" => "/^(\d{5})$/",
         "FO" => "/^(?:FO)*(\d{3})$/",
         "FR" => "/^(\d{5})$/",
-        "GB" => "/^(([A-Z]\d{2}[A-Z]{2})|([A-Z]\d{3}[A-Z]{2})|([A-Z]{2}\d{2}[A-Z]{2})|([A-Z]{2}\d{3}[A-Z]{2})|([A-Z]\d[A-Z]\d[A-Z]{2})|([A-Z]{2}\d[A-Z]\d[A-Z]{2})|(GIR0AA))$/",
+        "GB" => "/^([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([AZa-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z])))) [0-9][A-Za-z]{2})$/",
```

Comment:
I agree that the ```[AZa-z]``` is non-sense. I have sent a request to the UK gov using contact form to ask about this. I am sure it is an issue. I don't have any concrete example to test with and tell if it is really an error.

By the way I used https://regex101.com/ to validate the RegEx. It may help.